### ISSUE TITLE
Live reload seems doesn't work.

### ISSUE THREAD

**Issue by PoSayDone** [deviantfero/wpgtk#216]
Hi!
I've just installed wpgtk, and xsettingsd, but live reload int gtk apps doesn't work. Is it working out of the box, or I need to do some configurations?

**Comment by PoSayDone** [comment]
@jasperro script is working

**Comment by PoSayDone** [comment]
> Sorry for the late reply, could you point me to @jasperro's script?

```
import shutil
import subprocess
import os
import tempfile

"""Runs xsettingsd for a short time to reload gtk3 theme"""
if shutil.which("xsettingsd"):
    fd, path = tempfile.mkstemp()
    try:
        with os.fdopen(fd, 'w+') as tmp:
            tmp.write('Net/ThemeName "FlatColor"\n')
            tmp.close()
            subprocess.Popen.wait(subprocess.Popen(["timeout", "0.2s", "xsettingsd", "-c", path]))
    finally:
            os.remove(path)
```

**Comment by chrisJuresh** [comment]
@PoSayDone where does this go?

**Comment by PoSayDone** [comment]
> @PoSayDone where does this go?

Just make a .py file with this code, in wpgtk options mark "Run commandand after" and put it in "Command:" in wpgtk like this: "python /path/to/scrpit/reload.py"

**Comment by deviantfero** [comment]
This is exactly what `wpgtk` does, except with some extra steps, not sure why it's not working for you, here's a sample of what wpgtk does:

```
    # no settings daemon is running.
    # So GTK is getting theme info from gtkrc file
    # using xsettingd to set the same theme (parsing it from gtkrc)
    elif shutil.which("xsettingsd") and os.path.isfile(settings_ini):
        gtkrc = configparser.ConfigParser()
        gtkrc.read(settings_ini)

        if gtkrc["Settings"]:
            theme = gtkrc["Settings"].get("gtk-theme-name", "FlatColor")
            fd, path = tempfile.mkstemp()

            try:
                with os.fdopen(fd, "w+") as tmp:
                    tmp.write('Net/ThemeName "' + theme + '"\n')
                    tmp.close()
                    util.silent_call([
                        "timeout", "0.2s", "xsettingsd", "-c", path
                    ])
                logging.info(
                    "reloaded %s from settings.ini using xsettingsd"
                    % theme
                )
            finally:
                os.remove(path)
```

**Comment by PoSayDone** [comment]
> This is exactly what `wpgtk` does, except with some extra steps, not sure why it's not working for you, here's a sample of what wpgtk does:
> 
> ```
>     # no settings daemon is running.
>     # So GTK is getting theme info from gtkrc file
>     # using xsettingd to set the same theme (parsing it from gtkrc)
>     elif shutil.which("xsettingsd") and os.path.isfile(settings_ini):
>         gtkrc = configparser.ConfigParser()
>         gtkrc.read(settings_ini)
> 
>         if gtkrc["Settings"]:
>             theme = gtkrc["Settings"].get("gtk-theme-name", "FlatColor")
>             fd, path = tempfile.mkstemp()
> 
>             try:
>                 with os.fdopen(fd, "w+") as tmp:
>                     tmp.write('Net/ThemeName "' + theme + '"\n')
>                     tmp.close()
>                     util.silent_call([
>                         "timeout", "0.2s", "xsettingsd", "-c", path
>                     ])
>                 logging.info(
>                     "reloaded %s from settings.ini using xsettingsd"
>                     % theme
>                 )
>             finally:
>                 os.remove(path)
> ```
> 
> maybe you don't have a gtkrc file, I should default to trying to use `FlatColor` in that case

I've just sold my pc, maybe in one week i could check it

**Comment by deviantfero** [comment]
I've improved this fix in `6.2.4` update your wpgtk and you should not have to run this script anymore

**Comment by PoSayDone** [comment]
> I've improved this fix in `6.2.4` update your wpgtk and you should not have to run this script anymore

Thanks, would try it soon
